content/articles/a-look-at-c#-coding-conventions/index.md

```diff
@@ -0,0 +1,239 @@
+    public event Action EventProcessing;
+
+    // Implementation of a method
+    public void StartEventProcessing()
+    {
+        // Use of a local function
+        static int CountQueueItems() => WorkerQueue.Count;
+
+    }
+}
+```
+
+Since parameters are public properties of the record, a reader can use pascal casing when putting down positional accounts.
+
+```c#
+public record PublicIntitution
+(
+    string Country,
+    string County,
+    string Village,
+);
+```
+
+### Camel case
+The camelCasing convention, which is used for parameter names, capitalizes the first letter of each word except the first.
+
+```
+voidMain
+```
+
+Another implementation of the camel case is when naming private or internal fields.
+
+```
+camelCasing
+```
+
+```c#
+public class DataService
+{
+    private IWorkerQueue _workerQueue;
+}
+```
+
+It is also advisable to use the `s_prefix` and the `t_prefix` for the static fields respectively when dealing with thread and internal fields.
+
+```c#
+public class DataService
+{
+    private static IWorkerQueue s_workerQueue;
+
+    [ThreadStatic]
+    private static TimeSpan t_timeSpan;
+}
+```
+
+Camel casing is used when executing most of the method parameters.
+
+```c#
+public T SomeMethod<T>(int someNumber, bool isValid)
+{
+}
+```
+
+Apart from the above conventions, there are other several conventions that a programmer needs to consider and apply while in their daily routine of programming.
+1. The choice of word - Use readable identifier names. Underscores, hyphens, and other non-alphanumeric characters should not be used. The choice of word implies that not any word can be used when programming, instead the proper words should be utilized.
+2. Hungarian notation is neither acceptable nor encouraged.
+A hungarian notation is whereby the name of a variable or function denotes its intent or kind, as well as its type in some languages. This therefore creates a programming language dependency and makes maintenance more difficult.
+3. The use of Abbreviations and Acronyms - Do not use acronyms that aren't generally recognized, and even if they are, only use them when essential.
+4. Abbreviations and contractions should not be used as identifier names. This simply means that when one is writing an identifier, words should not be contracted but instead well-known identifiers should be used.
+5. Do not use `GetWin` instead of `GetWindow`. This is because it is somehow confusing instead a programmer is advised to use well-known abbreviations eg For UserInterface, one may use UI which is more identifiable.
+
+### Layout Conventions
+For a clear understanding of the code and its structure, formatting is exercised as a coding convention.
+
+Some of the layout coding conventions are:
+- Use the code editor's default settings that use smart indenting, four-character indents, and tabs that are saved as spaces.
+- Each line should accommodate one statement.
+- Each line should only accommodate one declaration.
```

Comment:
explain this two further. why should we accommodate one statement and why we each line should accommodate one declaration?

Reply:
One statement per line means one is not supposed to explain or to comment about two or more different fields on a single line. Each commenting line should consist a single explanation about a certain say variable, identifier or such.

One declaration per line simply means that declaring multiple variables in a single declaration may bring about confusion about the type of variables and their initial values. This will help the reader to have a smooth flow when reading.